src/util/util.c

```diff
@@ -204,9 +204,16 @@ static int getarg(int argc, char **argv)
         if (strlen(search_key) == 0)
                 usage(GETARG, EXIT_FAILURE, "search key undefined");
 
+        cmdline = strdup(p);
```

Comment:
Hmm, I see no reason to copy `p` in the first place... The function could just use it directly, i.e., initialize `cmdline` (the iterator) simply as `cmdline = p`. Or have I overlooked something?

Reply:
I have... `next_arg()` modifies the string, so it _must_ be copied. So scratch this.